### DISCUSSION TITLE
Slow UI response under MacOS.

### DISCUSSION THREAD

**Discussion by gbumgard** [SauceLLC/sauce4zwift#67] [General]
After updating the app a few weeks ago, it has become almost unusable on my Macbook Pro. Click and drag on individual widget windows is very difficult if not impossible. It can take several seconds after right clicking on a widget before the title bar appears, if it appears at all, and most attempts to drag or resize a widget simply results in the disappearance of the title bar and no response to the drag or resize operation.
Response time to keystrokes also lags. Using the keyboard to switch the data fields in the currently watching widget often lags by many seconds and sometimes seems to react as if the key was pressed multiple times.
This inability to move and resize windows is especially troublesome, as the widget window positions and sizes often changes from one session to the next. I believe this is somehow related to the use of multiple monitors as it most often occurs when starting Sauce after turning on an external display that had been off.
The app was sometimes slow to respond in the first version I installed, but now is much worse.
EDIT: it appears there is a relatively constant ~10 sec delay between mouse click or keypress and any response from the Sauce UI.

**Comment by mayfield** [comment]
Sounds like something is very broken.   I can't imagine any system spec where it would be that slow.  I don't want you to have to post system info here, so can you email me some screencaps to support@sauce.llc of Activity Monitor / CPU when Sauce is running and also with one of the Sauce windows can you hit Option+Command+I and capture the Console tab.
